api/src/main/java/javax/json/Json.java

```diff
@@ -57,9 +63,57 @@
  * threads.
  */
 public final class Json {
+    private static final Map<ClassLoader, JsonProvider> providers = Collections.synchronizedMap(new WeakHashMap<>());
 
     private Json() {
     }
+    
+    /**
+     * Sets the provider instance used by this factory class. The provider returned is associated with the context 
+     * class loader for the current thread (if one exists).
+     * 
+     * <p>Clients are encouraged to use this method to cache the desired provider instance, thus preventing 
+     * unnecessary ServiceLoader interactions (which may be costly).
+     * 
+     * @param provider a JsonProvider instance
+     */
+    public static void setProvider(JsonProvider provider) {
+        providers.put(getContextClassLoader(), Objects.requireNonNull(provider));
```

Comment:
This is new method in the *javax.* namespace. @m0mus are you really sure we are allowed to do that?

Reply:
I agree with @lukasj.  It is JSON-P 1.1.  Would we still call it "1.1" after changing it?

Related to this issue, the release of JSON-P and the reference implementation combined right now under one version.  It is a little awkward for us since we have to do constant 3rd party version updates against JSON-P even though we only depend on javax.json and nothing has changed.

Reply:
@jjspiegel I should link [this readme](https://github.com/eclipse-ee4j/jsonp/blob/master/bundles/ri/src/main/resources/README.txt) from some more prominent place...